posts/hello-world-machine.md

```diff
@@ -0,0 +1,252 @@
+
+The usual set up is a function (class/module) named `hello` with an included string of "Goodbye, Mars!" to be returned with a string "Hello, World!"
+
+```python
+def hello():
+    return 'Goodbye, Mars!'
+```
+
+Next, I will attempt to solve this quandary using various methods that I could think of. And for narrative purpose, I will refer to this simple function as *the* MACHINE!! whose only job is returning a greeting of : "Hello, World!". Let's see where it goes.
+
+
+#### The Traditional
+
+```python
+def hello() -> str:
+    return 'Hello, World'
+```
+
+With just a simple change of string from "Goodbye, Mars!" to "Hello, World!", we can now safely return it.
+
+#### The Parameter
+
+You decide maybe it is a good idea to give the machine an input of the phrase, so it can remember it for later usage.
+
+```python
+def hello(greeting: str) -> str:
+    return greeting
+
+hello("Hello, World!")
+```
+
+#### The Named Parameter
+
+After the previous attemp, the machine only managed to remember where is it but it can't remember the greeting, so you need to put in the greeting again.
+
+```python
+def hello(greeting: str, location="World") -> str:
+    return greeting + ", " + location + "!"
+
+hello("Hello")
+```
+
+### The -join
+
+The machine already have some of the words, but it can't seen to string a ~~sen-~~ string together.
+
+```python
+def hello() -> str:
+    words = ["Hello", " world!"]
+    return ",".join(words)
+```
+
+#### The Title
+
+The machine's battery is a bit low today, all the words are lacking energy and not being capitalized properly.
+
+```python
+def hello() -> str:
+    words = "hello, world!"
+    return words.title()
+```
+
+#### The Reversal
+
+The machine somehow have the greeting ready, but something about it looks a bit ..backward.
+
+```python
+def hello() -> str:
+    sdrow = "!dlroW ,olleH"
+    return sdrow[::-1]
+```
+**Note:** `Reverse String` is a featured exercise for the #48in24 challenge, remember to check out various approaches of how to do it in different languages!
+
+#### The Slice 'n Dice
+
+Maybe the machine need to have its oil changed, the normal greeting message seem to be a bit mixed up.
+
+```python
+def hello() -> str:
+	w = "Holld, Werlo!"
+	return f"{w[0]}{w[8]}{w[2:4]}{w[1]}{w[5:8]}{w[-2]}{w[-4:-2]}{w[4]}{w[-1]}"
+```
+
+
+#### The Curry
+
+Oh oh, some parts of the machine is in maintenance mode. It forgot the greeting  message and now can only accept part by part back.
+
+```python
+def hello(greet: str):
+    def world(location: str) -> str:
+        return f"{greet}, {location}!"
+    return world
+
+hello = hello("Hello")
+hello("World")
+```
+
+#### The Ascii
+
+You accidentally spilled some lemonade on the machine and now it only show you these strange number instead of the usual characters in the message.
```

Comment:
```suggestion
You accidentally spilled some lemonade on the machine and now it only shows you these strange numbers instead of the usual characters in the message.
```